### ISSUE TITLE
Manual libnuma install required

### ISSUE THREAD

**Issue by lbordowitz** [haskell/ghcup#58]
I ran the following commands:

```bash
$ ghcup install
$ ghcup set 8.4.4
$ ghcup install-cabal
$ cabal update
```

All of these commands executed successfully. However, I then ran this command, and got the following error:

```bash
$ cabal new-install cabal-install
...
Warning: Some package(s) failed to build. Try rerunning with -j1 if you can't
see the error.

Failed to build digest-0.0.1.2.
....
Configuring digest-0.0.1.2...
Preprocessing library for digest-0.0.1.2..
/usr/bin/ld.gold: error: cannot find -lnuma
```

After manually installing the library with `apt install libnuma-devel`, the `cabal-install` worked correctly. However, if I didn't know how to read complex error logs and understand what was missing, it would have taken a lot longer to figure out the problem.

```bash
$ uname -a
Linux [computer-name-here] 4.15.0-42-generic #45-Ubuntu SMP Thu Nov 15 19:32:57 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
```

**Comment by hvr** [comment]
This is the downside of the `ghcup` approach as compared to using system-packaged binaries which can declare their respective dependencies within their native packaging framework. See #49 for the meta-ticket concerning how to document bindist dependencies.

**Comment by hasufell** [comment]
Yes, but here lies a deeper issue even: GHC bindists for debian have numa support enabled at compile time, which even leaks into the rts and ALL your ghc compiled binaries. This is potentially dangerous for deployment, depending on your setup.

So I consider this a portability issue as well.

**Comment by alanz** [comment]
@hasufell the libnuma requirement is a regression in GHC, when creating the bindists for GHC 8.6.3

**Comment by hasufell** [comment]
Where can we raise issues about the official binaries?

**Comment by hvr** [comment]
@hasufell the proper place to bring up ghc bindist concerns is https://ghc.haskell.org/trac/ghc/newticket (but check first whether there's already an issue for it)

**Comment by hasufell** [comment]
> the libnuma requirement is a regression in GHC, when creating the bindists for GHC 8.6.3

https://ghc.haskell.org/trac/ghc/ticket/15688 this one?

**Comment by hvr** [comment]
@hasufell while that one's also about libnuma, it's not so much about the binary dists, but rather about a bug of the GHC build-system, i.e. different phase

**Comment by hasufell** [comment]
Also see https://ghc.haskell.org/trac/ghc/ticket/15444 linked from [reddit thread](https://www.reddit.com/r/haskell/comments/an405v/ghc_up_installation_fails_with_error/)

**Comment by hasufell** [comment]
> adding the capacity for ghcup to install extra system libs

Never.

**Comment by gbaz** [comment]
ok, i'm not pressing for it, just raising the question. in the interest of being able to answer this if someone asks me why: what do i tell them?

**Comment by hasufell** [comment]
> ok, i'm not pressing for it, just raising the question. in the interest of being able to answer this if someone asks me why: what do i tell them?

It's not unix philosophy. It's terrible. It's taking away control of the user, invoking semi-random sudo commands that might or might not do what is correct or trash half of your system, because you now have sudo rights?

Even if you would say "screw unix", then how would you support all distros with different configurations? If I think about gentoo alone then the complexity explodes. In addition, ghcup tries to be non-interactive.

**Comment by chaosrun** [comment]
I have another similar situation, I don't know if it's related, but it solved by install some lib package, I just share a solution here in case others have the same issue (when I google the error, this issue page appears):

**System Ubuntu 18.04**

`Linux <host_name> 4.15.0-46-generic #49-Ubuntu SMP Wed Feb 6 09:33:07 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux`

```bash
curl https://raw.githubusercontent.com/haskell/ghcup/master/bootstrap-haskell -sSf | sh
```

**Stdout**

```
[220 of 220] Compiling Distribution.Simple ( Distribution/Simple.hs, dist/build/Distribution/Simple.o )
/usr/bin/ld.gold: error: cannot find -lgmp
collect2: error: ld returned 1 exit status
`gcc' failed in phase `Linker'. (Exit code: 1)
Warning: Some package(s) failed to build. Try rerunning with -j1 if you can't
see the error.
cabal: Failed to build Cabal-2.4.1.0 (which is required by
cabal-install-2.4.1.0).
```

**Install libgmp3-dev**

```bash
sudo apt update && sudo apt install libgmp3-dev -y
```

**Stdout**

Try agian:

```
Configuring digest-0.0.1.2...
cabal: Missing dependency on a foreign library:
* Missing (or bad) header file: zlib.h
* Missing (or bad) C library: z
```

**Install zlib1g-dev**

```bash
sudo apt isntall zlib1g-dev -y
```

**Stdout**

Try again:

```
Configuring digest-0.0.1.2...
Preprocessing library for digest-0.0.1.2..
/usr/bin/ld.gold: error: cannot find -lnuma
collect2: error: ld returned 1 exit status
linking dist/build/Data/Digest/CRC32_hsc_make.o failed (exit code 1)
```

**Install libnuma-dev**

```bash
sudo apt install libnuma-dev -y
```

**Stdout**

Try agian:

```
Installation done!
```

**Comment by hasufell** [comment]
We should start documenting these requirements upstream at ghc gitlab: https://gitlab.haskell.org/ghc/ghc/wikis/index

**Comment by hasufell** [comment]
We currently have `ghcup print-system-reqs`, but these are not per ghc. I'm not sure it's worthwhile to make these per GHC, given hat the libnuma thing was a bug.
